.emacs.d/lisp/visual/line-numbers-c.el

```diff
@@ -0,0 +1,44 @@
+(defun kotct/line-numbers--set-up-linum ()
+  "Sets up `linum-mode' to run globally except in a few cases."
+  (progn
+    ;; enable linum globally
+    (global-linum-mode 1)
+
+    ;; use a lambda to generate the linum formatting
+    ;; this will use the minimum number of columns,
+    ;; but also always right-align
+    (setq linum-format (lambda (line)
+                         (let* ((w (length (number-to-string (count-lines (point-min) (point-max)))))
+                                (thing (concat " %" (number-to-string (max 2 w)) "d ")))
+                           (propertize (format thing line) 'face 'linum))))
```

Comment:
not for this PR, but I think this lambda is causing a lot of performance issues in weird cases

Reply:
I do too.